src/main/java/edu/harvard/iq/dataverse/api/Access.java

```diff
@@ -1829,192 +1807,72 @@ private boolean isAccessAuthorized(DataFile df, String apiToken) {
         } else {
             logger.fine("Session is null.");
         } 
-        
-        User apiTokenUser = null;
-        
-        if ((apiToken != null)&&(apiToken.length()!=64)) {
-            // We'll also try to obtain the user information from the API token, 
-            // if supplied: 
-        
-            try {
-                logger.fine("calling apiTokenUser = findUserOrDie()...");
-                apiTokenUser = findUserOrDie();
-            } catch (WrappedResponse wr) {
-                logger.log(Level.FINE, "Message from findUserOrDie(): {0}", wr.getMessage());
-            }
-            
-            if (apiTokenUser == null) {
-                logger.warning("API token-based auth: Unable to find a user with the API token provided.");
-            }
+        //If we don't have a user, nothing more to do. (Note session could have returned GuestUser)
+        if (user == null && apiTokenUser == null) {
+            logger.warning("Unable to find a user via session or with a token.");
```

Comment:
Looking in the old code - checking for the key length to not be 64 chars is very old and seems to be backwards even if there was a time the keys were 64 chars. Regardless, since the Access class doesn't need to check the token any more (all handled in the AbstractApiBean.findUserOrDie method), this isn't needed anymore.

Reply:
Those 64 characters-long tokens were used by GeoConnect/Worldmap; that system needed token-based authentication, back when our standard REST API did not support tokens. So that subsystem had its own tokens implemented and we had to support them for a while, for backward compatibility. I understand that we got rid of everything worldmap-related in the code base... (but, just in case, somebody correct me if I'm wrong!)